### ISSUE TITLE
terraform plan -destroy doesnt destroy a resource

### ISSUE THREAD

**Issue by gtmtech** [hashicorp/terraform#4452]
I have a remote state file (in s3). When executing terraform show, I can clearly see I have a resource:

aws_instance.foo

(indeed the instance is up and running).

I have removed the code for this resource from my local .tf files, as I no longer need it. Finally the last stage is to clean it up, which I normally do with a separate plan -destroy, then apply

terraform plan -destroy -target=aws_instance.foo

```
Refreshing Terraform state prior to plan...


No changes. Infrastructure is up-to-date. This means that Terraform
could not detect any differences between your configuration and
the real physical resources that exist. As a result, Terraform
doesn't need to do anything.
```

Is this incorrect behaviour? - surely it should be trying to destroy the resource.

**Comment by gtmtech** [comment]
Terraform 0.6.6

**Comment by jen20** [comment]
Hi @gtmtech - I believe with Terraform 0.6.6 you're hitting the issue which was fixed in #3912 and released in Terraform 0.6.7. Could you try upgrading the version of Terraform you are using, and reopen this issue if it does not resolve the problem? Thanks!
